utils/src/main/java/com/nedap/archie/datetime/DateTimeSerializerFormatters.java

```diff
@@ -56,22 +58,31 @@ public static String serializeDuration(TemporalAmount amount) {
             return serializeDuration((Duration) amount);
         } else if (amount instanceof Period) {
             return serializeDuration((Period) amount);
+        } else if (amount instanceof PeriodDuration) {
+            return serializePeriodDuration((PeriodDuration) amount);
         }
         return amount.toString();
     }
 
     private static String serializeDuration(Duration duration) {
         if(duration.isNegative()) {
-            return "-" + Duration.ZERO.minus(duration).toString();
+            return "-" + duration.negated();
         }
         return duration.toString();
     }
 
     private static String serializeDuration(Period period) {
         if(period.isNegative()) {
-            return "-" + Period.ZERO.minus(period).toString();
+            return "-" + period.negated();
         }
         return period.toString();
     }
 
+    private static String serializePeriodDuration(PeriodDuration periodDuration) {
+        if(periodDuration.getPeriod().isNegative() || periodDuration.getDuration().isNegative()) {
+            return "-" + periodDuration.negated();
+        }
+        return periodDuration.toString();
+    }
```

Comment:
These could use some test, especially `PeriodDuration` with combined positive and negative parts.

Reply:
I'll add some tests, even though this of course is already covered with the XML+JSON+ADL tests.
OpenEHR does not support durations with combined negative and positive parts, and I'm not sure if we should support them here. Converting them is non-trivial, especially with periods.
An example: P1Y-1D can be expressed as a Period, but not in openEHR. Converting it is tricky: 1year-1day can only be expressed as a number of days, or a number of weeks plus days. But depending on whether that is a leap-year, the exact number of days changes. So it's probably better to not support this?
Durations are exact, and much easier to do.
If we ever need this, I would suggest a separate issue.

Reply:
If it is not supported in openEHR, then I agree not to support it here.